eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/utils/factory/Lazy.java

```diff
@@ -26,6 +26,10 @@
 {
     /**
      * Returns the instance which is created on first access.
+     * <p>
+     * Note:
+     * If <code>T</code> is mutable, the returned object is always the same instance.
+     * If <code>T</code> is immutable, the instance can be different (as there is no need for it to be the same).
```

Comment:
These two notes are pointless. The important point is that the "identity" of the result is the same on every call. It doesn't matter if the result is mutable or not. I think the definition of `Lazy` already makes clear how it behaves. Besides, this method doesn't get any parameters, so the result is expected to be the same on every call (since the `Lazy` instance is supposed to be immutable).